File: src/main/java/com/github/redfoos/logstash/formatting/LogstashElementTypeSets.java
package com.github.redfoos.logstash.formatting;

import com.intellij.psi.tree.TokenSet;

import static com.github.redfoos.logstash.psi.LogstashTypes.*;

public interface LogstashElementTypeSets {

    TokenSet LOGSTASH_SAME_INDENT = TokenSet.create(
        LBRACE,
        RBRACE,
        LBRACKET,
        RBRACKET,
        LPARENTH,
        RPARENTH,
        COMMENT
    );

    TokenSet AROUND_SPACES_TOKENS = TokenSet.create(
        EQUALS,
        LESS_THAN,
        GREATER_THAN,
        GREATER_THAN_OR_EQUAL,
        LESS_THAN_OR_EQUAL,
        REGEXP_EQUAL,
        REGEXP_NOT_EQUAL,
        RIGHT_ARROW,
        AND_OPERATOR,
        IN_OPERATOR,
        NAND_OPERATOR,
        AND_OPERATOR,
        OR_OPERATOR,
        XOR_OPERATOR
    );
}

File: src/main/kotlin/com/github/redfoos/logstash/formatting/LogstashElementTypeSets.kt
package com.github.redfoos.logstash.formatting

import com.github.redfoos.logstash.psi.LogstashTypes
import com.intellij.psi.tree.TokenSet

interface LogstashElementTypeSets {
    companion object {
        val LOGSTASH_SAME_INDENT = TokenSet.create(
            LogstashTypes.LBRACE,
            LogstashTypes.RBRACE,
            LogstashTypes.LBRACKET,
            LogstashTypes.RBRACKET,
            LogstashTypes.LPARENTH,
            LogstashTypes.RPARENTH,
        )

        val AROUND_SPACES_TOKENS = TokenSet.create(
            LogstashTypes.EQUALS,
            LogstashTypes.LESS_THAN,
            LogstashTypes.GREATER_THAN,
            LogstashTypes.GREATER_THAN_OR_EQUAL,
            LogstashTypes.LESS_THAN_OR_EQUAL,
            LogstashTypes.REGEXP_EQUAL,
            LogstashTypes.REGEXP_NOT_EQUAL,
            LogstashTypes.RIGHT_ARROW,
            LogstashTypes.AND_OPERATOR,
            LogstashTypes.IN_OPERATOR,
            LogstashTypes.NAND_OPERATOR,
            LogstashTypes.AND_OPERATOR,
            LogstashTypes.OR_OPERATOR,
            LogstashTypes.XOR_OPERATOR
        )
    }
}
